From 58e6ed8f61f2d6e2d55dcb77ac9e5f71c8c516af Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 14 Sep 2018 11:42:14 +0800
Subject: [PATCH 4636/5242] MLK-19584 gpu: imx: imx8_dprc: Improve code for
 dprc_configure()

commit  c48ba115256f6513fb8a01d83861a260977d0443 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch contains code-change-only to improve code for
dprc_configure().  No real logic modification.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |   21 +++++++++------------
 1 file changed, 9 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index 46aa84e..c5e7eba 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -549,20 +549,17 @@ void dprc_configure(struct dprc *dprc, unsigned int stream_id,
 	}
 	dprc_write(dprc, val, MODE_CTRL0);
 
-	if (start || dprc->is_blit_chan) {
+	if (dprc->is_blit_chan) {
+		val = SW_SHADOW_LOAD_SEL | RUN_EN | SHADOW_LOAD_EN;
+		dprc_write(dprc, val, SYSTEM_CTRL0);
+	} else if (start) {
 		/* software shadow load for the first frame */
-		val = SW_SHADOW_LOAD_SEL;
-		if (dprc->is_blit_chan) {
-			val |= RUN_EN | SHADOW_LOAD_EN;
-			dprc_write(dprc, val, SYSTEM_CTRL0);
-		} else {
-			val |= SHADOW_LOAD_EN;
-			dprc_write(dprc, val, SYSTEM_CTRL0);
+		val = SW_SHADOW_LOAD_SEL | SHADOW_LOAD_EN;
+		dprc_write(dprc, val, SYSTEM_CTRL0);
 
-			/* and then, run... */
-			val |= RUN_EN | REPEAT_EN;
-			dprc_write(dprc, val, SYSTEM_CTRL0);
-		}
+		/* and then, run... */
+		val |= RUN_EN | REPEAT_EN;
+		dprc_write(dprc, val, SYSTEM_CTRL0);
 	}
 
 	prg_configure(dprc->prgs[0], width, height, x_offset, y_offset,
-- 
1.7.9.5

